web/packages/shared/components/UnifiedResources/shared/guessAppIcon.ts

```diff
@@ -0,0 +1,117 @@
+/**
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {
+  ResourceIconName,
+  resourceIconSpecs,
+  iconNames,
+} from 'design/ResourceIcon/resourceIconSpecs';
+
+import { UnifiedResourceApp } from '../types';
+
+export function guessAppIcon(resource: UnifiedResourceApp): ResourceIconName {
+  const { awsConsole = false, name, friendlyName, labels } = resource;
+
+  if (awsConsole) {
+    return 'aws';
+  }
+
+  // Label matching takes precedence and we can assume it can be a direct lookup
+  // since we expect a certain format.
+  const labelIconValue = labels?.find(l => l.name === 'teleport.icon')?.value;
+  if (labelIconValue && resourceIconSpecs[labelIconValue]) {
+    return labelIconValue as ResourceIconName;
+  }
```

Comment:
@avatus i couldn't remember our discussion, what did we want to name these again?

Reply:
i think `teleport.icon` is fine as it dictates what it is "icon" and that it's from teleport. we can't use teleport-internal since that gets filtered out. This will work IMO